flink-ml-lib/src/main/java/org/apache/flink/ml/feature/standardscaler/OnlineStandardScalerModel.java

```diff
@@ -0,0 +1,298 @@
+        private final RowTypeInfo inputTypeInfo;
+
+        private final String inputCol;
+
+        private final boolean withMean;
+
+        private final boolean withStd;
+
+        private final long maxAllowedModelDelayMs;
+
+        private final String modelVersionCol;
+
+        private ListState<StreamRecord> bufferedPointsState;
+
+        private ListState<StandardScalerModelData> modelDataState;
+
+        /** Model data for inference. */
+        private StandardScalerModelData modelData;
+
+        private DenseVector mean;
+
+        /** Inverse of standard deviation. */
+        private DenseVector scale;
+
+        private long modelVersion;
+
+        private long modelTimeStamp;
+
+        public PredictionOperator(
+                RowTypeInfo inputTypeInfo,
+                String inputCol,
+                boolean withMean,
+                boolean withStd,
+                long maxAllowedModelDelayMs,
+                String modelVersionCol) {
+            this.inputTypeInfo = inputTypeInfo;
+            this.inputCol = inputCol;
+            this.withMean = withMean;
+            this.withStd = withStd;
+            this.maxAllowedModelDelayMs = maxAllowedModelDelayMs;
+            this.modelVersionCol = modelVersionCol;
+        }
+
+        @Override
+        public void initializeState(StateInitializationContext context) throws Exception {
+            super.initializeState(context);
+            bufferedPointsState =
+                    context.getOperatorStateStore()
+                            .getListState(
+                                    new ListStateDescriptor<StreamRecord>(
+                                            "bufferedPoints",
+                                            new StreamElementSerializer(
+                                                    inputTypeInfo.createSerializer(
+                                                            getExecutionConfig()))));
+
+            modelDataState =
+                    context.getOperatorStateStore()
+                            .getListState(
+                                    new ListStateDescriptor<>(
+                                            "modelData",
+                                            TypeInformation.of(StandardScalerModelData.class)));
+        }
+
+        @Override
+        public void snapshotState(StateSnapshotContext context) throws Exception {
+            super.snapshotState(context);
+            if (modelData != null) {
+                modelDataState.clear();
+                modelDataState.add(modelData);
+            }
+        }
+
+        @Override
+        public void open() throws Exception {
+            super.open();
+            MetricGroup mlModelMetricGroup =
+                    getRuntimeContext()
+                            .getMetricGroup()
+                            .addGroup(MLMetrics.ML_GROUP)
+                            .addGroup(MLMetrics.ML_MODEL_GROUP);
```

Comment:
Since we might have multiple models running in the same Flink Job, it might be better to specify the model name to reduce chance of conflict.

How about using `addGroup(MLMetrics.ML_MODEL_GROUP, "OnlineStandardScalerModel")` here?